packages/contracts/tests/QF/02-grantRound.ts

```diff
@@ -152,31 +256,205 @@ describe("Quadratic Funding Infrastructure New Grant Round", () => {
+        const expectedTimeStamp = timeStamp + index + 1;
+
+        return expect(qfi.connect(user).signUp(_pubKey, _signUpGatekeeperData, _initialVoiceCreditProxyData))
+          .to.emit(qfi, "SignUp")
+          .withArgs(expectedStateIndex, expectedPubKey, expectedIntialVoiceCredits, expectedTimeStamp);
+      });
+      await Promise.all(signups);
+    });
+
+    it("verify - allows many users to sign up to maci with correct parameters", async () => {
+      const userSigners = [
+        user1,
+        user2,
+        user3,
+        user4,
+        user5,
+        user6,
+        user7,
+        user8,
+        user9,
+        user10,
+        user11,
+        user12,
+        user13,
+        user14,
+        user15,
+        user16,
+        user17,
+        user18,
+        user19,
+        user20,
+        user21,
+        user22,
+        user23,
+        user24,
+        user25,
+      ];
+      for (const user of userSigners) {
+        const maciKey = new Keypair();
+        const _pubKey = maciKey.pubKey.asContractParam();
+        const _signUpGatekeeperData = ethers.utils.defaultAbiCoder.encode(["uint256"], [1]);
+        const _initialVoiceCreditProxyData = ethers.utils.defaultAbiCoder.encode(["uint256"], [0]);
+        await expect(qfi.connect(user).signUp(_pubKey, _signUpGatekeeperData, _initialVoiceCreditProxyData)).to.emit(
+          qfi,
+          "SignUp"
+        );
+      }
+    });
+
+    it("require fail - sign up fails with invalid keys", async () => {
+      const _badPubKey = {
+        x: "21888242871839275222246405745257275088548364400416034343698204186575808495617",
+        y: "0",
+      };
+      const _signUpGatekeeperData = ethers.utils.defaultAbiCoder.encode(["uint256"], [1]);
+      const _initialVoiceCreditProxyData = ethers.utils.defaultAbiCoder.encode(["uint256"], [0]);
+
+      await expect(
+        qfi.connect(user4).signUp(_badPubKey, _signUpGatekeeperData, _initialVoiceCreditProxyData)
+      ).to.be.revertedWith("MACI: _pubKey values should be less than the snark scalar field");
+    });
+  });
+  describe("Create a new Poll", () => {
+    const _duration = 15;
+    const _maxValues = {
+      maxUsers: 25,
+      maxMessages: 25,
+      maxVoteOptions: 25,
+    };
+    const _treeDepths = {
+      intStateTreeDepth: 1,
+      messageTreeDepth: 4,
+      messageTreeSubDepth: 2,
+      voteOptionTreeDepth: 2,
+    };
+    const _intStateTreeDepth = 1;
+    const _messageBatchSize = 25;
+    const _tallyBatchSize = 5 ** _intStateTreeDepth;
+    coordinator = new Keypair();
+    const _coordinatorPubkey = coordinator.pubKey.asContractParam();
+
+    const expectedPollId = "0";
+    //const expectedPollAddress = "0x4aAa546748B9b4DBaa3A1C58f0DD1b2d3C6E97Ef"; // NOTE: nonce hack to get the poll address. Use this for Integration test
+    const expectedPollAddress = "0xB550ae8163A275BD1a43D2074716FC384c696265"; // NOTE: nonce hack to get the poll address. Use this for Unit test
```

Comment:
I think the nonce workaround isn't working correctly, currently getting `0xb22F17625EC8fD0173098Ae0C04485FD3F11BD19`